Shared/Samples/Browse WFS layers/BrowseWFSLayersView.swift

```diff
@@ -0,0 +1,186 @@
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+/// The initial view of the Browse WFS Layers sample.
+struct BrowseWFSLayersView: View {
+    /// The URL of the service to load.
+    @State private var serviceURL = URL(string: "https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&request=getcapabilities")!
+    /// A Boolean value indicating whether the service view is pushed on the
+    /// navigation stack.
+    @State private var isServiceViewPresented = false
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    TextField(
+                        "WFS Service",
+                        value: $serviceURL,
+                        format: .url,
+                        prompt: Text("WFS Service URL")
+                    )
+                    Button("Load") {
+                        isServiceViewPresented = true
+                    }
+                    .foregroundStyle(.accent)
+                    .containerRelativeFrame(.horizontal)
+                    .multilineTextAlignment(.center)
+                }
+            }
+            .navigationDestination(isPresented: $isServiceViewPresented) {
+                WFSServiceView(serviceURL: serviceURL)
+            }
+        }
+    }
+}
+
+/// A view that loads a WFS service and lists its layers.
+private struct WFSServiceView: View {
+    /// The URL of the WFS service to load.
+    let serviceURL: URL
+    
+    @Environment(\.dismiss) private var dismiss
+    
+    /// The loaded WFS service.
+    @State private var service: WFSService?
+    /// The error if the service failed to load, otherwise `nil`.
+    @State private var serviceLoadError: Error?
+    
+    var body: some View {
+        if let serviceInfo = service?.serviceInfo {
+            Form {
+                Section("Layers") {
+                    ForEach(serviceInfo.layerInfos, id: \.name) { layerInfo in
+                        NavigationLink(layerInfo.title) {
+                            WFSServiceLayerView(layerInfo: layerInfo)
+                        }
+                    }
+                }
+            }
+            .navigationTitle(serviceInfo.title)
+        } else {
+            ProgressView()
+                .navigationTitle("Loading Service")
+                .task {
+                    let service = WFSService(url: serviceURL)
+                    do {
+                        try await withTaskCancellationHandler {
+                            try await service.load()
+                        } onCancel: {
+                            service.cancelLoad()
+                        }
+                        self.service = service
+                    } catch {
+                        serviceLoadError = error
+                    }
+                }
+                .errorAlert(presentingError: $serviceLoadError)
```

Comment:
The alert modifier you previously used dismisses the view when OK is tapped. Do you still want to keep that behavior? If not, the `DismissAction` can be removed.

Reply:
Oh that's right! I did want that. All right, I'll revert my previous commit and just set the state so that the alert gets presented.